File: exercises/ruby_more_topics/easy_02/02.rb
# Exercise 02

# Write your own version of zip that does the same type of operation. 
# It should take two Arrays as arguments, and return a new Array 
# (the original Arrays should not be changed). Do not use the built-in 
# Array#zip method. You may assume that both input arrays have the 
# same number of elements.

def zip(array_a, array_b)
  output_array = []

  (0...array_a.size).each do |index|
    output_array << [array_a[index], array_b[index]]
  end

  output_array
end

File: exercises/js_small_problems/easy_04/02.js
// Exercise 02

// Write a function that takes two arrays as arguments and returns 
// an array containing the union of the values from the two. 
// There should be no duplication of values in the returned array, 
// even if there are duplicates in the original arrays. 
// You may assume that both arguments will always be arrays.



function union(array1, array2) {
  return [...new Set([...array1, ...array2])];
}


console.log(union([1, 3], [1]));